### ISSUE TITLE
$ sign shouldn't work in <pre> in md files

### ISSUE THREAD

**Issue by AndreWin** [sagemathinc/smc#112]
For example I need to write the following code:
```
$ cd ~
$ mkdir apps
```
This code block contains $ signs and is converted to \<pre\> in preview side. This code in \<pre\> should be shown as it was typed, but Mathjax tries to parse content in this code block.

You can see more information about it [here](https://groups.google.com/forum/#!topic/sage-cloud/JAwrzv7XS3k)

Best regards, Andrey.

**Comment by williamstein** [comment]
Not parsing $'s in pre environments would save us a lot of pain in a lot of places!!

**Comment by haraldschilly** [comment]
so, uhm, in my eyes this ticket is a hard one. however.

1. I can't reproduce the originally reported problem. I've tested single or double dollars in chats inside a pre, in a markdown file, and in `%md` in a sagews. everything fine. Hence, even interpreting this more broadly, this bug is ok. 

2. Despite that, I assume @williamstein  has tested this to be still the case 4 days ago, so I'm definitely missing some corner cases.

3. a case where I know something is off is ```x `$abc` y``` an  a chat. I boiled this down to how markdown processing is done in `markdown.coffee:markdown_to_html`. I've submitted PR #954 to address this detail.

4. the only real undecidable case in markdown is this: 

        $a^b
        $xxx

   where there is no newline between the two and they're also not in `pre` or ```. It's somehow automatically converted to being a  one-liner and hence the heuristic in `misc.parse_mathjax` no longer detects it.
